fdbserver/workloads/GetMappedRange.actor.cpp

```diff
@@ -47,9 +48,13 @@ struct GetMappedRangeWorkload : ApiWorkload {
 	//	const bool SPLIT_RECORDS = deterministicRandom()->random01() < 0.5;
 	const bool SPLIT_RECORDS = true;
 	const static int SPLIT_SIZE = 3;
+	int checkStorageQueueSeconds;
+	int queueMaxLength;
```

Comment:
Can you change both to double?

M1 CI found an error because of this:
```
GetMappedRange.actor.g.cpp
In file included from /Users/ec2-user/foundationdb/fdbserver/workloads/GetMappedRange.actor.cpp:22:
In file included from /Users/ec2-user/foundationdb/fdbclient/include/fdbclient/MutationLogReader.actor.h:25:
In file included from /Users/ec2-user/foundationdb/fdbclient/include/fdbclient/MutationLogReader.actor.h:30:
In file included from /Users/ec2-user/foundationdb/fdbclient/include/fdbclient/FDBTypes.h:35:
In file included from /Users/ec2-user/foundationdb/fdbclient/include/fdbclient/Status.h:24:
In file included from /Users/ec2-user/foundationdb/fdbclient/include/fdbclient/JSONDoc.h:23:
In file included from /Users/ec2-user/foundationdb/fdbclient/include/fdbclient/json_spirit/json_spirit_writer_template.h:13:
/Users/ec2-user/foundationdb/fdbclient/include/fdbclient/json_spirit/json_spirit_value.h:485:9: error: no matching function for call to 'get_value'
        return internal_::get_value(*this, internal_::Type_to_type<T>());
               ^~~~~~~~~~~~~~~~~~~~
/Users/ec2-user/foundationdb/fdbclient/include/fdbclient/JSONDoc.h:268:17: note: in instantiation of function template specialization 'json_spirit::Value_impl<json_spirit::Config_map<std::string>>::get_value<long>' requested here
                        out = pLast->get_value<T>();
                                     ^
/Users/ec2-user/foundationdb/fdbserver/workloads/GetMappedRange.actor.cpp:484:12: note: in instantiation of function template specialization 'JSONDoc::get<long>' requested here
                                                role.get("query_queue_max", queryQueueMax);
                                                     ^
/Users/ec2-user/foundationdb_build_output_macos_arm64/fdbserver/workloads/GetMappedRange.actor.g.cpp:3399:16: note: in instantiation of member function 'GetMappedRangeWorkload::ReportMetricActorState<GetMappedRangeWorkload::ReportMetricActor>::a_body1loopBody1cont5' requested here
                        loopDepth = a_body1loopBody1cont5(loopDepth);
                                    ^
/Users/ec2-user/foundationdb_build_output_macos_arm64/fdbserver/workloads/GetMappedRange.actor.g.cpp:3272:16: note: in instantiation of member function 'GetMappedRangeWorkload::ReportMetricActorState<GetMappedRangeWorkload::ReportMetricActor>::a_body1loopBody1cont2' requested here
                        loopDepth = a_body1loopBody1cont2(loopDepth);
                                    ^
/Users/ec2-user/foundationdb_build_output_macos_arm64/fdbserver/workloads/GetMappedRange.actor.g.cpp:3316:15: note: in instantiation of member function 'GetMappedRangeWorkload::ReportMetricActorState<GetMappedRangeWorkload::ReportMetricActor>::a_body1loopBody1cont1' requested here
                loopDepth = a_body1loopBody1cont1(result, loopDepth);
                            ^
/Users/ec2-user/foundationdb_build_output_macos_arm64/fdbserver/workloads/GetMappedRange.actor.g.cpp:3231:151: note: in instantiation of member function 'GetMappedRangeWorkload::ReportMetricActorState<GetMappedRangeWorkload::ReportMetricActor>::a_body1loopBody1when1' requested here
                if (__when_expr_0.isReady()) { if (__when_expr_0.isError()) return a_body1Catch1(__when_expr_0.getError(), std::max(0, loopDepth - 1)); else return a_body1loopBody1when1(__when_expr_0.get(), loopDepth); };
                                                                                                                                                                    ^
/Users/ec2-user/foundationdb_build_output_macos_arm64/fdbserver/workloads/GetMappedRange.actor.g.cpp:3220:49: note: in instantiation of member function 'GetMappedRangeWorkload::ReportMetricActorState<GetMappedRangeWorkload::ReportMetricActor>::a_body1loopBody1' requested here
                while (loopDepth == oldLoopDepth) loopDepth = a_body1loopBody1(loopDepth);
                                                              ^
/Users/ec2-user/foundationdb_build_output_macos_arm64/fdbserver/workloads/GetMappedRange.actor.g.cpp:3199:16: note: in instantiation of member function 'GetMappedRangeWorkload::ReportMetricActorState<GetMappedRangeWorkload::ReportMetricActor>::a_body1loopHead1' requested here
                        loopDepth = a_body1loopHead1(loopDepth);
                                    ^
/Users/ec2-user/foundationdb_build_output_macos_arm64/fdbserver/workloads/GetMappedRange.actor.g.cpp:3720:9: note: in instantiation of member function 'GetMappedRangeWorkload::ReportMetricActorState<GetMappedRangeWorkload::ReportMetricActor>::a_body1' requested here
                this->a_body1();
                      ^
/Users/ec2-user/foundationdb/fdbclient/include/fdbclient/json_spirit/json_spirit_value.h:442:5: note: candidate function template not viable: no known conversion from 'Type_to_type<long>' to 'Type_to_type<int>' for 2nd argument
int get_value(const Value& value, Type_to_type<int>) {
    ^
/Users/ec2-user/foundationdb/fdbclient/include/fdbclient/json_spirit/json_spirit_value.h:447:9: note: candidate function template not viable: no known conversion from 'Type_to_type<long>' to 'Type_to_type<int64_t>' for 2nd argument
int64_t get_value(const Value& value, Type_to_type<int64_t>) {
        ^
/Users/ec2-user/foundationdb/fdbclient/include/fdbclient/json_spirit/json_spirit_value.h:452:10: note: candidate function template not viable: no known conversion from 'Type_to_type<long>' to 'Type_to_type<uint64_t>' for 2nd argument
uint64_t get_value(const Value& value, Type_to_type<uint64_t>) {
         ^
/Users/ec2-user/foundationdb/fdbclient/include/fdbclient/json_spirit/json_spirit_value.h:457:8: note: candidate function template not viable: no known conversion from 'Type_to_type<long>' to 'Type_to_type<double>' for 2nd argument
double get_value(const Value& value, Type_to_type<double>) {
       ^
/Users/ec2-user/foundationdb/fdbclient/include/fdbclient/json_spirit/json_spirit_value.h:462:29: note: candidate function [with Value = json_spirit::Value_impl<json_spirit::Config_map<std::string>>] not viable: no known conversion from 'Type_to_type<long>' to 'Type_to_type<typename Value_impl<Config_map<string>>::String_type>' for 2nd argument
typename Value::String_type get_value(const Value& value, Type_to_type<typename Value::String_type>) {
                            ^
/Users/ec2-user/foundationdb/fdbclient/include/fdbclient/json_spirit/json_spirit_value.h:467:23: note: candidate function [with Value = json_spirit::Value_impl<json_spirit::Config_map<std::string>>] not viable: no known conversion from 'Type_to_type<long>' to 'Type_to_type<typename Value_impl<Config_map<string>>::Array>' for 2nd argument
typename Value::Array get_value(const Value& value, Type_to_type<typename Value::Array>) {
                      ^
/Users/ec2-user/foundationdb/fdbclient/include/fdbclient/json_spirit/json_spirit_value.h:472:24: note: candidate function [with Value = json_spirit::Value_impl<json_spirit::Config_map<std::string>>] not viable: no known conversion from 'Type_to_type<long>' to 'Type_to_type<typename Value_impl<Config_map<string>>::Object>' for 2nd argument
typename Value::Object get_value(const Value& value, Type_to_type<typename Value::Object>) {
                       ^
/Users/ec2-user/foundationdb/fdbclient/include/fdbclient/json_spirit/json_spirit_value.h:477:6: note: candidate function template not viable: no known conversion from 'Type_to_type<long>' to 'Type_to_type<bool>' for 2nd argument
bool get_value(const Value& value, Type_to_type<bool>) {
     ^
1 error generated.
```

Reply:
> Can you change both to double?
> 
> M1 CI found an error because of this:
> 
> ```
> GetMappedRange.actor.g.cpp
> In file included from /Users/ec2-user/foundationdb/fdbserver/workloads/GetMappedRange.actor.cpp:22:
> In file included from /Users/ec2-user/foundationdb/fdbclient/include/fdbclient/MutationLogReader.actor.h:25:
> In file included from /Users/ec2-user/foundationdb/fdbclient/include/fdbclient/MutationLogReader.actor.h:30:
> In file included from /Users/ec2-user/foundationdb/fdbclient/include/fdbclient/FDBTypes.h:35:
> In file included from /Users/ec2-user/foundationdb/fdbclient/include/fdbclient/Status.h:24:
> In file included from /Users/ec2-user/foundationdb/fdbclient/include/fdbclient/JSONDoc.h:23:
> In file included from /Users/ec2-user/foundationdb/fdbclient/include/fdbclient/json_spirit/json_spirit_writer_template.h:13:
> /Users/ec2-user/foundationdb/fdbclient/include/fdbclient/json_spirit/json_spirit_value.h:485:9: error: no matching function for call to 'get_value'
>         return internal_::get_value(*this, internal_::Type_to_type<T>());
>                ^~~~~~~~~~~~~~~~~~~~
> /Users/ec2-user/foundationdb/fdbclient/include/fdbclient/JSONDoc.h:268:17: note: in instantiation of function template specialization 'json_spirit::Value_impl<json_spirit::Config_map<std::string>>::get_value<long>' requested here
>                         out = pLast->get_value<T>();
>                                      ^
> /Users/ec2-user/foundationdb/fdbserver/workloads/GetMappedRange.actor.cpp:484:12: note: in instantiation of function template specialization 'JSONDoc::get<long>' requested here
>                                                 role.get("query_queue_max", queryQueueMax);
>                                                      ^
> /Users/ec2-user/foundationdb_build_output_macos_arm64/fdbserver/workloads/GetMappedRange.actor.g.cpp:3399:16: note: in instantiation of member function 'GetMappedRangeWorkload::ReportMetricActorState<GetMappedRangeWorkload::ReportMetricActor>::a_body1loopBody1cont5' requested here
>                         loopDepth = a_body1loopBody1cont5(loopDepth);
>                                     ^
> /Users/ec2-user/foundationdb_build_output_macos_arm64/fdbserver/workloads/GetMappedRange.actor.g.cpp:3272:16: note: in instantiation of member function 'GetMappedRangeWorkload::ReportMetricActorState<GetMappedRangeWorkload::ReportMetricActor>::a_body1loopBody1cont2' requested here
>                         loopDepth = a_body1loopBody1cont2(loopDepth);
>                                     ^
> /Users/ec2-user/foundationdb_build_output_macos_arm64/fdbserver/workloads/GetMappedRange.actor.g.cpp:3316:15: note: in instantiation of member function 'GetMappedRangeWorkload::ReportMetricActorState<GetMappedRangeWorkload::ReportMetricActor>::a_body1loopBody1cont1' requested here
>                 loopDepth = a_body1loopBody1cont1(result, loopDepth);
>                             ^
> /Users/ec2-user/foundationdb_build_output_macos_arm64/fdbserver/workloads/GetMappedRange.actor.g.cpp:3231:151: note: in instantiation of member function 'GetMappedRangeWorkload::ReportMetricActorState<GetMappedRangeWorkload::ReportMetricActor>::a_body1loopBody1when1' requested here
>                 if (__when_expr_0.isReady()) { if (__when_expr_0.isError()) return a_body1Catch1(__when_expr_0.getError(), std::max(0, loopDepth - 1)); else return a_body1loopBody1when1(__when_expr_0.get(), loopDepth); };
>                                                                                                                                                                     ^
> /Users/ec2-user/foundationdb_build_output_macos_arm64/fdbserver/workloads/GetMappedRange.actor.g.cpp:3220:49: note: in instantiation of member function 'GetMappedRangeWorkload::ReportMetricActorState<GetMappedRangeWorkload::ReportMetricActor>::a_body1loopBody1' requested here
>                 while (loopDepth == oldLoopDepth) loopDepth = a_body1loopBody1(loopDepth);
>                                                               ^
> /Users/ec2-user/foundationdb_build_output_macos_arm64/fdbserver/workloads/GetMappedRange.actor.g.cpp:3199:16: note: in instantiation of member function 'GetMappedRangeWorkload::ReportMetricActorState<GetMappedRangeWorkload::ReportMetricActor>::a_body1loopHead1' requested here
>                         loopDepth = a_body1loopHead1(loopDepth);
>                                     ^
> /Users/ec2-user/foundationdb_build_output_macos_arm64/fdbserver/workloads/GetMappedRange.actor.g.cpp:3720:9: note: in instantiation of member function 'GetMappedRangeWorkload::ReportMetricActorState<GetMappedRangeWorkload::ReportMetricActor>::a_body1' requested here
>                 this->a_body1();
>                       ^
> /Users/ec2-user/foundationdb/fdbclient/include/fdbclient/json_spirit/json_spirit_value.h:442:5: note: candidate function template not viable: no known conversion from 'Type_to_type<long>' to 'Type_to_type<int>' for 2nd argument
> int get_value(const Value& value, Type_to_type<int>) {
>     ^
> /Users/ec2-user/foundationdb/fdbclient/include/fdbclient/json_spirit/json_spirit_value.h:447:9: note: candidate function template not viable: no known conversion from 'Type_to_type<long>' to 'Type_to_type<int64_t>' for 2nd argument
> int64_t get_value(const Value& value, Type_to_type<int64_t>) {
>         ^
> /Users/ec2-user/foundationdb/fdbclient/include/fdbclient/json_spirit/json_spirit_value.h:452:10: note: candidate function template not viable: no known conversion from 'Type_to_type<long>' to 'Type_to_type<uint64_t>' for 2nd argument
> uint64_t get_value(const Value& value, Type_to_type<uint64_t>) {
>          ^
> /Users/ec2-user/foundationdb/fdbclient/include/fdbclient/json_spirit/json_spirit_value.h:457:8: note: candidate function template not viable: no known conversion from 'Type_to_type<long>' to 'Type_to_type<double>' for 2nd argument
> double get_value(const Value& value, Type_to_type<double>) {
>        ^
> /Users/ec2-user/foundationdb/fdbclient/include/fdbclient/json_spirit/json_spirit_value.h:462:29: note: candidate function [with Value = json_spirit::Value_impl<json_spirit::Config_map<std::string>>] not viable: no known conversion from 'Type_to_type<long>' to 'Type_to_type<typename Value_impl<Config_map<string>>::String_type>' for 2nd argument
> typename Value::String_type get_value(const Value& value, Type_to_type<typename Value::String_type>) {
>                             ^
> /Users/ec2-user/foundationdb/fdbclient/include/fdbclient/json_spirit/json_spirit_value.h:467:23: note: candidate function [with Value = json_spirit::Value_impl<json_spirit::Config_map<std::string>>] not viable: no known conversion from 'Type_to_type<long>' to 'Type_to_type<typename Value_impl<Config_map<string>>::Array>' for 2nd argument
> typename Value::Array get_value(const Value& value, Type_to_type<typename Value::Array>) {
>                       ^
> /Users/ec2-user/foundationdb/fdbclient/include/fdbclient/json_spirit/json_spirit_value.h:472:24: note: candidate function [with Value = json_spirit::Value_impl<json_spirit::Config_map<std::string>>] not viable: no known conversion from 'Type_to_type<long>' to 'Type_to_type<typename Value_impl<Config_map<string>>::Object>' for 2nd argument
> typename Value::Object get_value(const Value& value, Type_to_type<typename Value::Object>) {
>                        ^
> /Users/ec2-user/foundationdb/fdbclient/include/fdbclient/json_spirit/json_spirit_value.h:477:6: note: candidate function template not viable: no known conversion from 'Type_to_type<long>' to 'Type_to_type<bool>' for 2nd argument
> bool get_value(const Value& value, Type_to_type<bool>) {
>      ^
> 1 error generated.
> ```

it seems the issue is at line `role.get("query_queue_max", queryQueueMax);`, but I can change them both to `double` after fixing the issue anyway